plugins/modules/dcnm_policy.py

```diff
@@ -0,0 +1,1072 @@
+
+        sw_dict = config.pop(pos)
+
+        # 'switches' contaions the switches related configuration items from playbook. Process these
+        # and add the same to individual policy items as appropriate
+
+        new_config = []
+        for sw in sw_dict['switch']:
+
+            # Check if policies are included in the switch config. If so these policies will override
+            # all other polcy configs given in the playbook globally. If not the switch can be included 
+            # in all individaul policy items
+
+            if (sw.get ('policies', None) != None):
+
+                # 'policies are specified at switch level. Add eachj of these policies to 'config' object 
+                # along with the switch information
+
+                for pol in sw['policies']:
+
+                    if (pol.get('switch', None) == None):
+                        pol['switch'] = []
+                    if (sw['ip'] not in pol['switch']):
+                        pol['switch'].append(sw['ip'])
+                    #if (pol not in new_config):
+                    new_config.append(pol)
+            else:
+
+                # This switch does not have any policies included. Add this switch to all policies in the 
+                # playbook config
+
+                for cfg in config:
+
+                    if (cfg.get('switch', None) == None):
+                        cfg['switch'] = []
+                    if (sw['ip'] not in cfg['switch']):
+                        cfg['switch'].append(sw['ip'])
+        if (new_config != []):                 
+            config.extend(new_config)
+
+        return config
+
+def main():
+
+    """ main entry point for module execution
+    """
+    element_spec = dict(
+        fabric=dict(required=True, type='str'),
+        config=dict(required=False, type='list'),
+        state=dict(type='str', default='merged',
+                   choices = ['merged', 'deleted', 'query']),
+        deploy=dict(required=False, type='bool', default=True)
+    )
+
+    with open ("check.log", "w+") as f:
+        f.write ("Entered, ES = {}\n".format(element_spec))
+    module = AnsibleModule(argument_spec=element_spec,
+                           supports_check_mode=True)
+
+    dcnm_policy = DcnmPolicy(module)
+
+    # Note down the global 'deploy' status. We will have to check this and the local 'deploy' flags 
+    # included with individual policies to decide if a policy is to be deployed or not.
+    dcnm_policy.deploy = module.params['deploy']
+    
+    start = datetime.datetime.now()
+
+    if not dcnm_policy.ip_sn:
+        dcnm_policy.result['msg'] = "Fabric {} missing on DCNM or does not have any switches".format(dcnm_policy.fabric)
+        module.fail_json(msg="Fabric {} missing on DCNM or does not have any switches".format(dcnm_policy.fabric))
+
+    state = module.params['state']
+    if not dcnm_policy.config:
+        if state == 'merged' or state == 'deleted' or state == 'query':
```

Comment:
I can't remember what we discussed.  Did we decide that `config` was mandatory for `deleted` state?  This means we cannot omit `config` and in order to delete all policies?

Reply:
Yes. Because there are many policies on the server and we cannot delete all. So we decided that the user must explicitly state which policies need to be deleted